Cell Modem Status Message

benjaminaltieri opened the pull request:
This new message facilitates the reporting of cell modem signal levels. As other information may be beneficial I have looked at a few standard commands and added fields accordingly:

AT+CSQ -> signal_strength, signal_error_rate
AT+CCID -> sim_id
AT+COPS? -> network operator (if present)
AT+CPIN?/AT+CREG?/AT+CGREG? - sim_status (roughly mapped to no sim/unregistered/registered on network)

As this may not be exhaustive, an unspecified field as a variable length array allows for extra length left at the end for as-of-yet unknown fields.

If an argument is made to only include what I currently need for cell signal reporting, I am totally fine doing that as well.

mfine commented on spec/yaml/swiftnav/sbp/piksi.yaml:
```diff
@@ -523,6 +523,38 @@ definitions:
+      - sim_status:
+          type: string
+          desc: Current status of sim whether inserted, ready, etc.
+      - unspecified:
```
We've historically used `reserved` here for these fields not currently in use.

benjaminaltieri replied:
Ok cool I'll probably just axe the other fields for now in the interest of getting this message available for the base functionality

mfine commented on spec/yaml/swiftnav/sbp/piksi.yaml:
```diff
@@ -523,6 +523,38 @@ definitions:
+          size: 28
+          desc: Currently reported network operator
+      - sim_status:
+          type: string
```
Sending the message here with unsized string and an array fill won't work - they're both greedy.

benjaminaltieri replied:
Good catch thanks!

benjaminaltieri commented:
I've removed the currently unused fields and switched the last floating parameter to `reserved` as suggested.

Generated the products and all tests pass. Final approval anyone?
